server/sonar-web/src/main/js/components/SourceViewer/components/DuplicationPopup.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import { Link } from 'react-router';
+import { groupBy, sortBy } from 'lodash';
+import { SourceViewerFile, DuplicationBlock, DuplicatedFile } from '../../../app/types';
+import BubblePopup from '../../common/BubblePopup';
+import QualifierIcon from '../../shared/QualifierIcon';
+import { translate } from '../../../helpers/l10n';
+import { collapsedDirFromPath, fileFromPath } from '../../../helpers/path';
+import { getProjectUrl } from '../../../helpers/urls';
+
+interface Props {
+  blocks: DuplicationBlock[];
+  // TODO use branchLike
+  branch: string | undefined;
+  duplicatedFiles?: { [ref: string]: DuplicatedFile };
+  inRemovedComponent: boolean;
+  onClose: () => void;
+  popupPosition?: any;
+  sourceViewerFile: SourceViewerFile;
+}
+
+export default class DuplicationPopup extends React.PureComponent<Props> {
+  isDifferentComponent = (
+    a: { project: string; subProject?: string },
+    b: { project: string; subProject?: string }
+  ) => {
+    return Boolean(a && b && (a.project !== b.project || a.subProject !== b.subProject));
+  };
+
+  handleFileClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
+    event.preventDefault();
+    event.currentTarget.blur();
+    const Workspace = require('../../workspace/main').default;
+    const { key, line } = event.currentTarget.dataset;
+    Workspace.openComponent({ key, line, branch: this.props.branch });
+    this.props.onClose();
+  };
+
+  render() {
+    const { duplicatedFiles = {}, sourceViewerFile } = this.props;
+
+    const groupedBlocks = groupBy(this.props.blocks, '_ref');
+    let duplications = Object.keys(groupedBlocks).map(fileRef => {
+      return {
+        blocks: groupedBlocks[fileRef],
+        file: duplicatedFiles[fileRef]
+      };
+    });
+    duplications = sortBy(duplications, d => {
+      const a = d.file.projectName !== sourceViewerFile.projectName;
+      const b = d.file.subProjectName !== sourceViewerFile.subProjectName;
+      const c = d.file.key !== sourceViewerFile.key;
+      return '' + a + b + c;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Use a template literal instead of concatenating with a string literal. ([typescript:S3512](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3512))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Fpopups&issues=AWHb6qaEw6XGHKvPuQTP&open=AWHb6qaEw6XGHKvPuQTP)

[ISSUE_KEY]: AWHb6qaEw6XGHKvPuQTP